### ISSUE TITLE
click.exceptions.BadParameter:  is not a valid boolean when incorrect configs are in place

### ISSUE THREAD

**Issue by Mno-hime** [cloudsmith-io/cloudsmith-cli#17]
I follow the `README.md` document step-by-step and once sample configuration files are mentioned I errorneously copied them to their respective directories:

```bash
 newman  ~  curl https://raw.githubusercontent.com/cloudsmith-io/cloudsmith-cli/master/cloudsmith_cli/data/credentials.ini > .config/cloudsmith/credentials.ini
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   501  100   501    0     0    533      0 --:--:-- --:--:-- --:--:--   532

 newman  ~  curl https://raw.githubusercontent.com/cloudsmith-io/cloudsmith-cli/master/cloudsmith_cli/data/config.ini > .config/cloudsmith/config.ini
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   737  100   737    0     0    485      0  0:00:01  0:00:01 --:--:--   485^[[A
```

And I noticed that `cloudsmith` CLI immediatelly started to crash:

```python3
 newman  ~  cloudsmith token
Usage: cloudsmith [OPTIONS] COMMAND [ARGS]...

Traceback (most recent call last):
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 1134, in invoke
    Command.invoke(self, ctx)
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/newman/.local/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/newman/.local/lib/python3.7/site-packages/cloudsmith_cli/cli/decorators.py", line 86, in wrapper
    opts.load_config_file(path=config_file, profile=profile)
  File "/home/newman/.local/lib/python3.7/site-packages/cloudsmith_cli/cli/config.py", line 187, in load_config_file
    return config_cls.load_config(self, path, profile=profile)
  File "/home/newman/.local/lib/python3.7/site-packages/cloudsmith_cli/cli/config.py", line 117, in load_config
    config = cls.read_config()
  File "/home/newman/.local/lib/python3.7/site-packages/click_configfile.py", line 401, in read_config
    cls.process_config_section(config_section, storage)
  File "/home/newman/.local/lib/python3.7/site-packages/click_configfile.py", line 445, in process_config_section
    section_data = parse_config_section(config_section, schema)
  File "/home/newman/.local/lib/python3.7/site-packages/click_configfile.py", line 270, in parse_config_section
    value = param.parse(value)
  File "/home/newman/.local/lib/python3.7/site-packages/click_configfile.py", line 197, in parse
    return self.type.convert(text, self, ctx=None)
  File "/home/newman/.local/lib/python3.7/site-packages/click/types.py", line 355, in convert
    self.fail('%s is not a valid boolean' % value, param, ctx)
  File "/home/newman/.local/lib/python3.7/site-packages/click/types.py", line 69, in fail
    raise BadParameter(message, ctx=ctx, param=param)
click.exceptions.BadParameter:  is not a valid boolean

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/newman/.local/bin/cloudsmith", line 10, in <module>
    sys.exit(main())
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/newman/.local/lib/python3.7/site-packages/click/core.py", line 734, in main
    e.show()
  File "/home/newman/.local/lib/python3.7/site-packages/click/exceptions.py", line 70, in show
    echo('Error: %s' % self.format_message(), file=file, color=color)
  File "/home/newman/.local/lib/python3.7/site-packages/click/exceptions.py", line 101, in format_message
    param_hint = self.param.get_error_hint(self.ctx)
AttributeError: 'Param' object has no attribute 'get_error_hint'
```

I suggest two things: (1) catch the exception and make the error message in this case more sensible, (2) perhaps put the paragraph on sample config files after the `cloudsmith token` procedure.

I fixed my system by removing those sample config files and running `cloudsmith token`.

I am on Fedora 30 with Python 3.7 and `cloudsmith` 0.16 from PyPI via `pip-3`.

**Comment by paddycarey** [comment]
Hi @Mno-hime thanks for the detailed report! We'll look into this ASAP and issue updated instructions in the README.

Catching the exception is definitely a good idea, I'll check with the team to see if we can fix that in the very near future.

**Comment by lskillen** [comment]
Fixed now on master by #18 - Thank you @Mno-hime!
